src/jaxsim/parsers/rod/meshes.py

```diff
@@ -0,0 +1,194 @@
+    elif isinstance(obj, rod.builder.primitive_builder.PrimitiveBuilder):
+        raise NotImplementedError("PrimitiveBuilder not implemented")
+    else:
+        raise ValueError("Invalid object type")
+
+
+def extract_points_vertices(mesh) -> np.ndarray:
+    """
+    Extracts the vertices of a mesh as points."""
+    return mesh.vertices
+
+
+def extract_points_random_surface_sampling(mesh, n: int = -1) -> np.ndarray:
+    """
+    Extracts N random points from the surface of a mesh.
+
+    Args:
+        mesh: The mesh from which to extract points.
+        n: The number of points to extract. If -1, all vertices are extracted.
+
+    Returns:
+        The extracted points (N x 3 array).
+    """
+
+    if n > 0 and n <= len(mesh.vertices):
+        return mesh.sample(n)
+    else:
+        if n != -1:
+            logging.warning(
+                "Invalid number of points for random surface sampling. Defaulting to all vertices"
+            )
+        return mesh.vertices
+
+
+def extract_points_uniform_surface_sampling(mesh, n: int = -1) -> np.ndarray:
+    """
+    Extracts N uniformly sampled points from the surface of a mesh.
+
+    Args:
+        mesh: The mesh from which to extract points.
+        n: The number of points to extract. If -1, all vertices are extracted.
+
+    Returns:
+        The extracted points (N x 3 array).
+    """
+
+    if n > 0 and n <= len(mesh.vertices):
+        return trimesh.sample.sample_surface_even(mesh=mesh, count=n)
+    else:
+        if n != -1:
+            logging.warning(
+                "Invalid number of points for uniform surface sampling. Defaulting to all vertices"
+            )
+        return mesh.vertices
+
+
+def extract_points_select_points_over_axis(
+    mesh, axis: str, direction: str, n: int
+) -> np.ndarray:
+    """
+    Extracts N points from a mesh along a specified axis. The points are selected based on their position along the axis.
+
+    Args:
+        mesh: The mesh from which to extract points.
+        axis: The axis along which to extract points.
+        direction: The direction along the axis from which to extract points. Valid values are "higher" and "lower".
+        n: The number of points to extract.
+
+    Returns:
+        The extracted points (N x 3 array).
+    """
+
+    valid_dirs = ["higher", "lower"]
+    if direction not in valid_dirs:
+        raise ValueError(f"Invalid direction. Valid directions are {valid_dirs}")
+
+    arr = mesh.vertices
+
+    index = 0 if axis == "x" else 1 if axis == "y" else 2
+    # Sort the array in ascending order
+    sorted_arr = arr[arr[:, index].argsort()]
+
+    if direction == "lower":
+        # Select first N points
+        points = sorted_arr[:n]
+    elif direction == "higher":
+        # Select last N points
+        points = sorted_arr[-n:]
+    else:
+        raise ValueError(
+            f"Invalid direction {direction} for SelectPointsOverAxis method"
+        )
+
+    return points
```

Comment:
As a general rule, it's better to avoid chained conditionals for performance, and for that hashtables (dictionaries) are great
```suggestion
    dirs = {"higher": np.s_[:n], "lower": np.s_[-n:]}

    arr = mesh.vertices

    index = dict(zip(("x", "y", "z"), np.arange(3)))
    
    # Sort the array in ascending order
    arr.sort(axis=index[axis])

    return sorted_arr[dirs[direction]]
```